internal/elasticsearch/index/index.go

```diff
@@ -390,6 +390,13 @@ func ResourceIndex() *schema.Resource {
 			ForceNew:     true, // To add a tokenizer, the index must be closed, updated, and then reopened; we can't handle that here.
 			ValidateFunc: validation.StringIsJSON,
 		},
+		"analysis_char_filter": {
+			Type:         schema.TypeString,
+			Description:  "A JSON string describing the char_filters applied to the index.",
+			Optional:     true,
+			ForceNew:     true, // To add a char_filters, the index must be closed, updated, and then reopened; we can't handle that here.
```

Comment:
I'm not a big fan of the `ForceNew` here. 

Best case is someone gets to easily recreate a new index they're playing around with, worst case is they delete a large production index. Comparing that to the behaviour if we left `ForceNew: false`, the worst case is they have to manually `POST <index>/_close` and `_open` around the terraform apply. 

IMO there's a pathway for applying this without destroying the index, and destroying the index may be a pretty cataclysmic event. The error message is reasonably clear what the issue is, we should let the operators deal with it as best suits their use-case. 

```
Can't update non dynamic settings [[index.analysis.char_filter.zero_with_spaces.type, index.analysis.char_filter.zero_with_spaces.mappings]] for open indices [[test/9oUEz4DkTIKswrtbwrEWzg]]
```

Reply:
I agree.  Not only for this but I think we should have `deletion_protection` like cloud SQL or the other important resources have to prevent unintended deletion.
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance

Reply:
If we think it still would be better not to set `ForceNew`, let me create a separate PR because it should be also removed from the other fields too (like analysis_filter or analysis_normalizer).

Reply:
> Not only for this but I think we should have `deletion_protection`

Can you open an issue for this. 

>  let me create a separate PR because it should be also removed from the other fields too

+1 on this, but can we remove it from the new code in this PR first?

Reply:
Okay, then let me just put the change to this PR.
https://github.com/elastic/terraform-provider-elasticstack/pull/158/commits/4d26ec26771df592f56f4cb8799db918ac915e05